### PR TITLE
fixing cases related to defaulted machineset in aws- adding secur…

### PR DESCRIPTION
**Pull request by miyadav**
…ity grp

This is WIP , mostly the default values seems to have changed , will need modification on the polarion side as well and test on different profiles , will keep these changes , if too much effort will convert atleast  latest versions to golang .

### CODE REVIEW

**Comment by miyadav** [comment]
This fixes - https://issues.redhat.com/browse/OCPQE-7807
@sunzhaohua2 @huali9 @jhou1 PTAL .

Validation - https://mastern-jenkins-csb-openshift-qe.apps.ocp-c1.prod.psi.redhat.com/job/ocp-common/job/Runner/723907/console

File: features/machine/machine.feature [hunk 1]
```diff
@@ -386,19 +386,21 @@ Feature: Machine features testing
...
+      | ["spec"]["selector"]["matchLabels"]["machine.openshift.io/cluster-api-machineset"]                 | <name>                                                         |
+      | ["spec"]["template"]["metadata"]["labels"]["machine.openshift.io/cluster-api-cluster"]             | <%= infrastructure("cluster").infra_name %>                    |
+      | ["spec"]["template"]["spec"]["providerSpec"]["value"]["ami"]["id"]                                 | <%= cb.default_ami_id %>                                       |
+      | ["spec"]["template"]["spec"]["providerSpec"]["value"]["securityGroups"][0]["filters"][0]["values"] | <%= cb.default_secgrp[0].values[0].to_s.split(",")[1][11..-1]%>|
+      | ["spec"]["template"]["spec"]["providerSpec"]["value"]["placement"]["availabilityZone"]             | <%= cb.default_availability_zone %>                            |
+      | ["spec"]["template"]["spec"]["providerSpec"]["value"]["subnet"]["filters"][0]["values"]            | <%= cb.default_subnet[0].values[1] %>                          |
```
Comment (sunzhaohua2) [inline features/machine/machine.feature:401]: For the cluster if the machineset's subnet is in form of ID, will it fail?

Comment (sunzhaohua2) [reply features/machine/machine.feature:401]: Got it, thanks @miyadav we can merge this first to see if it still fails in some clusters. If it still fails, we can use golang. It is not easy to check whether the subnet is id or filter in cucushift.
It seems that when the vpc is not created by the installer, but create it ourselves, and then write the vpcid to install-config, the subnet will be in the form of id.  The template name contains customer-vpc, seems the subnet is id.

**Comment by miyadav** [comment]
@jhou1  Please help to merge , when time permits.

**Pull request merged**